src/tests/Microsoft.Agents.Auth.Tests/ConfigurationConnectionsTests.cs

```diff
@@ -0,0 +1,325 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Agents.Authentication;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Configuration.Memory;
+using Moq;
+using System;
+using System.Collections.Generic;
+using System.Runtime.Loader;
+using System.Security.Claims;
+using Xunit;
+
+namespace Microsoft.Agents.Auth.Tests
+{
+    public class ConfigurationConnectionsTests
+    {
+        private readonly Mock<IServiceProvider> _serviceProvider = new();
+        private readonly ClaimsIdentity _identity = new([]);
+
+        [Fact]
+        public void GetConnection_ShouldReturnAccessTokenProviderWithConnectionName()
+        {
+            var config = new ConfigurationRoot(
+            [
+                new MemoryConfigurationProvider(new MemoryConfigurationSource
+                {
+                    InitialData = new Dictionary<string, string>
+                    {
+                        { "ConnectionsMap:0:ServiceUrl", "*" },
+                        { "ConnectionsMap:0:Connection", "BotServiceConnection" },
+                        { "Connections:BotServiceConnection:Type", "MsalAuth" },
+                        { "Connections:BotServiceConnection:Assembly", "Microsoft.Agents.Authentication.Msal" },
+                        { "Connections:BotServiceConnection:Settings:ClientId", "ClientId" },
+                        { "Connections:BotServiceConnection:Settings:ClientSecret", "ClientSecret" },
+                        { "Connections:BotServiceConnection:Settings:AuthorityEndpoint", "AuthorityEndpoint" },
```

Comment:
maybe we can create a private property containing these configs? i see that not all these tests have the same settings, some use all, some use just BotServiceConnection, and others ServiceUrl, Connection, etc

Reply:
I thought that too, I tried, but they differ from test to test, so I left it as it was.